File: P5/prob5.rb
=begin
Author: Barrett Helms
Program: Problem 5, Project Euler
Purpose: Find the smallest positive number that is evenly divisible by 1-20
Date started: Mar-10-2013
Last modified: Mar-10-2013
Status: Solved
=end
count, total, num = 1, 1, 0
loop do
  if total % count == 0
    count += 1
    num = total
  else total += num end
  break if count > 20
end
p total

File: 1-9/P5/prob5.cpp
/*
Author: Barrett Helms
Program: Problem 5, Project Euler
Purpose: Find the smallest positive number that is evenly divisible by 1-20
Date started: Mar-10-2013
Date last modified: Mar-10-2013
Status: Solved
*/
#include <iostream>
int main()
{
    int num, count = 1, total = 1;
    while (count <= 20) {
        if (total % count == 0) {
            count++;
            num = total;
        }
        else {
            total += num;
        }
    }
    std::cout << total << std::endl;
    return 0;
}
